galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/PropertyRoute.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.routes;
+
+import static dev.galasa.framework.api.common.ServletErrorMessage.*;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Path;
+import java.util.Map;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.QueryParameters;
+import dev.galasa.framework.api.common.ResponseBuilder;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+
+/**
+ * An abstract route used by all the Property-related routes.
+ */
+public class PropertyRoute extends CPSRoute {
+
+    static final Gson gson = GalasaGsonBuilder.build();
+
+    // Define a default filter to accept everything
+    static DirectoryStream.Filter<Path> defaultFilter = path -> { return true; };
+
+
+    public PropertyRoute(ResponseBuilder responseBuilder, IFramework framework ) {
+		/* Regex to match endpoints: 
+		*  -> /cps/
+		*/
+		super(responseBuilder, "/cps/(.*)/properties/(.*)", framework);
+	}
+
+    protected IFramework getFramework() {
+        return super.framework;
```

Comment:
Odd that this class is returning something it fishes out from super(). Just move this method to super() and make it a private field ?